planning/fil_record_interpreter.go

```diff
@@ -0,0 +1,89 @@
+package planning
+
+import (
+	"errors"
+	"github.com/filecoin-project/indexer-reference-provider/metadata"
+	"github.com/filecoin-project/storetheindex/api/v0"
+	"github.com/ipfs-shipyard/w3rc/contentrouting"
+)
+
+// ------- FORK START --------------
+// TODO: Remove forked code below once #14 is merged.
+// Forked to unblock implementation.
+// See: https://github.com/ipfs-shipyard/w3rc/pull/14
+type (
+	PolicyName  string
+	PolicyScore float64
+	Policy      interface {
+		Name() PolicyName
+	}
+	PolicyResults interface {
+		// must be in range from zero to 1, will get dropped otherwise
+		// should return zero for policies that are unrecognized
+		Score(PolicyName) PolicyScore
+	}
+
+	FilecoinV1RecordInterpreter struct {
+	}
+
+	PreferFree struct{}
+)
+
+func (fp PreferFree) Name() PolicyName { return "prefer_free" }
+
+var _ Policy = PreferFree{}
+
+// ------- FORK END --------------
+
+func (fri *FilecoinV1RecordInterpreter) Interpret(record contentrouting.RoutingRecord, policies []Policy) (PolicyResults, error) {
+
+	// decode the record (or error) -- use metadata from filecoin
+	// check for free or paid policy
+	// return PolicyResults that when given "prefer_free" returns 1 if retrieval is free or zero if its paid
+
+	if record.Protocol() == contentrouting.RoutingErrorProtocol {
+		err, ok := record.Payload().(error)
+		if !ok {
+			return nil, errors.New("routing record payload not match expected type: error")
+		}
+		return nil, err
+	}
+
+	data, ok := record.Payload().([]byte)
+	if !ok {
+		return nil, errors.New("filecoin v1 routing record payload does not match expected type: []byte")
+	}
+	rm := v0.Metadata{
+		record.Protocol(),
+		data,
+	}
+
+	dtm, err := metadata.FromIndexerMetadata(rm)
+	if err != nil {
+		return nil, err
+	}
+
+	fv1d, err := metadata.DecodeFilecoinV1Data(dtm)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: How should policies argument impact the returned results?
```

Comment:
@hannahhoward  I would love your help in answering this question please.

Reply:
I agree. This is weird maybe we remove the policy list here?

Reply:
I am inclined to merge as is for now given that we are still developing code here.